doc/goals.rst

```diff
@@ -0,0 +1,159 @@
+
+A distinguishing feature of Dune is that it lets the user declare which
+version of the ``dune`` tool their project was written against, and
+``dune`` will morph itself to behave the same as this version of the
+``dune`` binary, even if it is a newer version. As a result, a recent
+version of the ``dune`` binary is able to understand a wide range of
+dune projects written against many different version of Dune. And
+while we strictly follow `semantic versioning`_, new major versions of
+Dune effectively introduce very few breaking changes and most projects
+do not need upper bounds on Dune.
+
+This guarantee is of course limited to documented behaviours.
+
+.. _semantic versioning: https://semver.org/
+
+Have a robust and scalable core
+===============================
+
+Tech companies tend be fond of big mono repositories, so to be able to
+work for them Dune needs to be able to eat up large repositories
+without blinking. It not only needs to build fast, but more
+importantly it must not get in the way of fast feedback during
+development, no matter the size of the repository.
+
+Note that Dune will only be tested on repositories as large as people
+participating in the development of Dune require. Currently, the
+largest user is Jane Street. If someone wanted to use Dune on much
+larger repositories than the ones used at Jane Street and this
+required a significant amount of effort on Dune, this wouldn't be
+considered unless we get some help to do so and we can keep the other
+promises.
+
+In particular, while making Dune scalable we must also make sure to
+not turn Dune into a monster because no one wants to force their users
+to install a monster to build their project. This brings us to the
+next point of Dune being a no-brainer dependency.
+
+Remain a no-brainer dependency
+==============================
+
+Dune is a hard dependency of any Dune project. Anyone using Dune
+to develop their project will have to ask their user to install
+Dune. For this reason, it is very important to keep Dune as lean as
+possible.
+
+This is why we have to be careful when we start relying on an external
+piece of software, or when we introduce new concepts. We must make
+sure to not introduce duplication or useless stuff. The overall
+projects has to remain lean.
+
+It is also important to keep Dune as easy to install as
+possible. Currently, the only requirement to build Dune is a
+working OCaml compiler. Nothing else is required, not even a shell and
+we should keep it this way.
+
+Remain accessible
+=================
+
+Since Dune aims to be the best possible tool for the whole OCaml
+community, it is important for Dune to remain accessible. Getting
+started and learning Dune should be straightforward.
+
+For that purpose, when designing the language, the command line
+interface or the documentation, we must always put ourselves from the
+point of view of a user who is discovering Dune and its features.
```

Comment:
For a slightly more idiomatic phrasing, we might put something like

```suggestion
interface or the documentation, we must take on the perspective 
of a user who is just discovering Dune and its features.
```

Reply:
Thanks, you version is more succinct, I like it better :)